test/metabase/driver/mysql_test.clj

```diff
@@ -27,12 +27,18 @@
    [metabase.test.data.interface :as tx]
    [metabase.util :as u]
    [metabase.util.date-2 :as u.date]
+   [metabase.util.honey-sql-2-extensions :as h2x]
+   #_{:clj-kondo/ignore [:discouraged-namespace]}
    [metabase.util.honeysql-extensions :as hx]
    [metabase.util.log :as log]
    [toucan.db :as db]
    [toucan.hydrate :refer [hydrate]]
    [toucan.util.test :as tt]))
 
+(use-fixtures :each (fn [thunk]
+                      (binding [hx/*honey-sql-version* 2]
+                        (thunk))))
```

Comment:
Stuff like `sql.qp/json-query` need to have `*honey-sql-version*` bound to work correctly. Normally the SQL QP will do this at the top level, but when we're testing internals we need to do it ourselves. Adding this fixture was easier than wrapping all the tests in this